packages/nc-gui/utils/formulaUtils.ts

```diff
@@ -416,6 +416,208 @@ const formulas: Record<string, any> = {
     syntax: 'WEEKDAY(date, [startDayOfWeek])',
     examples: ['WEEKDAY("2021-06-09")', 'WEEKDAY(NOW(), "sunday")'],
   },
+
+  TRUE: {
+    type: formulaTypes.NUMERIC,
+    validation: {
+      args: {
+        max: 0,
+      },
+    },
+    description: 'Returns 1',
+    syntax: 'TRUE()',
+    examples: ['TRUE()'],
+  },
+
+  FALSE: {
+    type: formulaTypes.NUMERIC,
+    validation: {
+      args: {
+        max: 0,
+      },
+    },
+    description: 'Returns 0',
+    syntax: 'FALSE()',
+    examples: ['FALSE()'],
+  },
+
+  REGEX_MATCH: {
+    type: formulaTypes.STRING,
+    validation: {
+      args: {
+        rqd: 2,
+      },
+    },
+    description: 'Returns 1 if the input text matches a regular expression or 0 if it does not.',
+    syntax: 'REGEX_MATCH(string, regex)',
+    examples: ['REGEX_MATCH({title}, "abc.*")'],
+  },
+
+  REGEX_EXTRACT: {
+    type: formulaTypes.STRING,
+    validation: {
+      args: {
+        rqd: 2,
+      },
+    },
+    description: 'Returns the first match of a regular expression in a string.',
+    syntax: 'REGEX_EXTRACT(string, regex)',
+    examples: ['REGEX_EXTRACT({title}, "abc.*")'],
+  },
+  REGEX_REPLACE: {
+    type: formulaTypes.STRING,
+    validation: {
+      args: {
+        rqd: 3,
+      },
+    },
+    description: 'Replaces all matches of a regular expression in a string with a replacement string',
+    syntax: 'REGEX_MATCH(string, regex, replacement)',
+    examples: ['REGEX_EXTRACT({title}, "abc.*", "abcd")'],
+  },
+  BLANK: {
+    type: formulaTypes.STRING,
+    validation: {
+      args: {
+        rqd: 0,
+      },
+    },
+    description: 'Returns a blank value(null)',
+    syntax: 'BLANK()',
+    examples: ['BLANK()'],
+  },
+  XOR: {
+    type: formulaTypes.NUMERIC,
```

Comment:
if we set it as `formulaTypes.NUMERIC`, we can input numbers (e.g. `XOR(1, 2, 3, {number})`) but which is not expected.

![image](https://github.com/nocodb/nocodb/assets/35857179/d6b7bf0e-3167-4241-9148-c8ec094b81cc)

Reply:
Can we set it to 0/1 ?

Reply:
will handled in next iteration :+1: